### PR TITLE
Add AzCLI and zip to dockerfile

### PR DESCRIPTION
**Pull request by mindlessroman**
PR adds documentation and the AzCLI and zip utilities to make the `env_setup.sh` run even more seamlessly.

### CODE REVIEW

**Review by polatengin — APPROVED** [review]
LGTM, thanks 🚀💯

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,6 +23,23 @@ Retrospectives is an Azure DevOps extension. Visit [this link](https://docs.micr
 3. Publish the newly created branch to the Retrospectives repo. Use this branch as your working branch.
 4. Once you are ready to check-in, create a pull request against the 'master' branch. Link the Bug/Task that you are fixing/adding to the pull request. Reviewers will be added automatically.
 
+## Considerations
```
Comment (argoldbe) [inline CONTRIBUTING.md]: @mindlessroman Adding the az cli & zip install to the Dockerfile is a good addition but I think the CONTRIBUTING.md changes may need to be reworked. This new section may give the impression that standing up a backend is required to contribute which is not the case. I like the list of dependencies (when not using the dev container) but would move to the "Backend" section. The "Backend" section may need some additional clarity on the fact that it's an optional requirement and is only needed for realtime updates. We can discuss more at stanup.

**Review by argoldbe — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,6 +23,23 @@ Retrospectives is an Azure DevOps extension. Visit [this link](https://docs.micr
...
+
+Provided in this repo is the environement setup script. Running `env_setup.sh` spins up inidivudal
+instances of the different Azure components for this extension. If you do not have all the packages
+and utilities installed already, it can slow down your development :(
+
+*The easiest solution?* This repo provides a Dockerfile and a devcontainer setup. This creates a simple,
+containerized development environment for contribuing to this extension. Follow along with [this video](https://www.youtube.com/watch?v=GwsscfcAjg8)
+by @polatengin that talks about the different ways to confirgure your environment.
```
Comment (NarmathaBala) [inline CONTRIBUTING.md]:
```suggestion
by @polatengin that talks about the different ways to configure your environment.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -22,6 +23,23 @@ Retrospectives is an Azure DevOps extension. Visit [this link](https://docs.micr
 3. Publish the newly created branch to the Retrospectives repo. Use this branch as your working branch.
 4. Once you are ready to check-in, create a pull request against the 'master' branch. Link the Bug/Task that you are fixing/adding to the pull request. Reviewers will be added automatically.
 
+## Considerations
+
+Provided in this repo is the environement setup script. Running `env_setup.sh` spins up inidivudal
+instances of the different Azure components for this extension. If you do not have all the packages
+and utilities installed already, it can slow down your development :(
+
+*The easiest solution?* This repo provides a Dockerfile and a devcontainer setup. This creates a simple,
```
Comment (NarmathaBala) [inline CONTRIBUTING.md]:
```suggestion
*The easiest solution* This repo provides a Dockerfile and a devcontainer setup. This creates a simple,
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -22,6 +23,23 @@ Retrospectives is an Azure DevOps extension. Visit [this link](https://docs.micr
 3. Publish the newly created branch to the Retrospectives repo. Use this branch as your working branch.
 4. Once you are ready to check-in, create a pull request against the 'master' branch. Link the Bug/Task that you are fixing/adding to the pull request. Reviewers will be added automatically.
 
+## Considerations
+
+Provided in this repo is the environement setup script. Running `env_setup.sh` spins up inidivudal
```
Comment (NarmathaBala) [inline CONTRIBUTING.md]:
```suggestion
Provided in this repo is the environment setup script. Running `env_setup.sh` spins up individual
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (NarmathaBala) [reply CONTRIBUTING.md]: +1 to language around env_setup.sh. Good to differentiate backend set up from frontend by adding sections for them.

**Review by NarmathaBala — CHANGES_REQUESTED** [review]
Thanks for making this contribution, @mindlessroman. I left a comment about differentiating backend setup from front end and suggestions.

File: CONTRIBUTING.md (hunk 1 above)
Comment (mindlessroman) [reply CONTRIBUTING.md]: Adding some context now

**Review by mhassaninmsft — APPROVED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (mindlessroman) [reply CONTRIBUTING.md]: I was on a role with those typos!

File: .devcontainer/Dockerfile [hunk 5]
```diff
@@ -20,6 +20,11 @@ RUN apt-get update \
     && apt-get update \
     && apt-get install -y docker-ce-cli \
     #
+    # Insall Azure CLI
```
Comment (focusthenflex) [inline .devcontainer/Dockerfile]:
```suggestion
    # Install Azure CLI

```

**Review by focusthenflex — APPROVED** [review]
LGTM

File: .devcontainer/Dockerfile (hunk 5 above)
Comment (mindlessroman) [reply .devcontainer/Dockerfile]: Great catch. All the more reason for a spell checker lolol

File: CONTRIBUTING.md (hunk 4 above)
Comment (focusthenflex) [reply CONTRIBUTING.md]: > I was on a role with those typos!

I was on a roll with those typos!

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -178,6 +192,12 @@ contains the App Service, App Service Plan and SignalR resources.
 CollaborationStateServiceUrl value to the App Service URL -
 `https://<RESOURCE_NAME_SUFFIX>.azurewebsites.net` and redeploy the extension.
 
+#### Considerations for the Backend
```
Comment (argoldbe) [inline CONTRIBUTING.md]: @mindlessroman I think it's safe to remove this "Considerations" section since you incorporated the details above.

**Review by argoldbe — APPROVED** [review]
Hannah - Thanks for making the changes we discussed. I left one small comment and then I think this is good to merge.

**Review by NarmathaBala — APPROVED** [review]
The changes look good now. Thanks @mindlessroman

**Pull request merged**